cookbooks/bcpc-hadoop/recipes/zabbix_web_monitoring.rb

```diff
@@ -0,0 +1,135 @@
+    if hostgroup_id.nil?
+      hostgroup_id = zbx.hostgroups.create(:name => "#{node.chef_environment}")
+
+      # Permission Guests usergroup to read hostgroup_id
+      guests_ugroup_id = zbx.usergroups.get_id(:name => 'Guests')
+      if guests_ugroup_id.nil?
+        Chef::Log.info("Could not find 'Guests' user group in zabbix")
+      else
+        ret = zbx.usergroups.set_perms(
+                :usrgrpid => guests_ugroup_id, :hostgroupids => [hostgroup_id],
+                :permission => 2)
+        if ret.nil?
+          Chef::Log.info("Failed to permission 'Guests' to read #{node.chef_environment}")
+        end
+      end
+    end #if hostgroup_id.nil?
+
+    # Create host for web moniroting
+    host_name="web_monitoring"
+    host_id = zbx.hosts.get_id(:host => "#{host_name}")
+    if host_id.nil?
+      host_id = zbx.hosts.create(
+        :host => "#{host_name}",
+        :interfaces => [{
+          :type => 1, :main => 1, :ip => '127.0.0.1', :dns => '127.0.0.1',
+          :port => 10050, :useip => 0
+        }],
+        :groups => [:groupid => "#{hostgroup_id}"]
+      )
+    end #if host_id.nil?
+
+    # iterate the web scenarios and create_or_update scenarios and related triggers and actions
+    scenarios = node.default[:bcpc][:hadoop][:zabbix][:web_monitoring][:scenarios] or {}
+    scenarios.each do | scenario |
+      Chef::Log.info("start processing web scenario #{scenario[:name]}")
+
+      # define names for the three parts
+      name_web     = "#{scenario[:name]}"
+      name_trigger = "#{scenario[:name]} trigger"
+      name_action  = "#{scenario[:name]} action"
+
+      # add scenario
+      zbx.httptests.create_or_update(
+        :name => name_web,
+        :delay => "15",
+        :retries=> "5",
+        :hostid => "#{host_id}",
+        :steps => [{
+          :no=>"1",
+          :name=> scenario[:name],
+          :url => scenario[:url],
+          :timeout=>"15",
+          :status_codes=>"200",
+          :follow_redirects=>"1",
+        }]
+      )
+
+      Chef::Log.info("added web scenario:" +
+        zbx.httptests.get_raw( { :filter => { :name => name_web } , :selectSteps => "extend" }).to_s)
+
+
+      # add triggers
+      trigger_expression = "{#{host_name}:#{scenario[:trigger_item]}[#{name_web}]." +
+        "#{scenario[:trigger_func]}}#{scenario[:trigger_cond]}"
+
+      zbx.triggers.create_or_update(
+        :description => name_trigger,
+        :expression => trigger_expression,
+        :comments => scenario[:description],
+        :hostid => host_id,
+        :priority => scenario[:severity],
+        :status => "0",
+        :type => "0"
+      )
+
+      Chef::Log.info("added trigger:" +
+        zbx.triggers.get_raw( { :filter => { :description => name_trigger }, :output => "extend", }).to_s)
+
+
+      # add action
+      action_status = node[:bcpc][:hadoop][:zabbix][:enable_alarming] ? 0 : 1 # status = 0 means enabled
```

Comment:
Actually I was mis-remembering how this worked and just looking for clarification. I was asking to clarify that one Chef run is necessary to change the triggers, if one is enabling or disabling alarming. But that is also how it works today...